andela/eagle-bn-backend
# #168781803 signup endpoint

#### What does this PR do?
Creates the signup endpoint 
#### Description of Task to be completed?
- create helper functions
- write tests
- create signup controller
- create signup route
- write documentation for signup
#### How should this be manually tested?
- clone the repo
- run npm start
- test /api/v1/users/signup endpoint:
 ```
parameters to be included in the request body :
- email
- password
- username
- confirmPassword
```

```
expected success body response:
{
    "msg": "string",
    "data": {
        "userId": "id of the user",
        "username": "username",
        "email": "lemail",
         "token": "JWT",
        "created_at": "Mon, 22 Oct 2018 00:00:00 GMT",
         "updated_at": "Mon, 22 Oct 2018 00:00:00 GMT" 
    }
}
```


#### Any background context you want to provide?
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/168781803
#### Screenshots (if appropriate)

#### Questions:
